src/Lucene.Net/Index/SegmentReader.cs

```diff
@@ -310,7 +317,10 @@ internal void  DecRef()
 			}
 			
 			internal void  OpenDocStores(SegmentInfo si, IState state)

```

Comment:
Was this method also affected (and you saw that in the stacktraces) or you modified it "just in case" because it might have the same issue?

Reply:
I added `Console.WriteLine` in all methods in order to see which ones are being used.
I don't see a reason to hold the lock if `fieldsReaderOrig != null`.